docs/zdgbook/ComponentsAndInterfaces.rst

```diff
@@ -258,8 +252,8 @@ ask an interface the names of all the various interface items it
 describes.  From the Python interpreter, for example, you can walk
 right up to an interface and ask it for its *names*::
 
-  >>> IUser.names()
-  ['getUserName', 'getFavoriteColor', 'getPassword']
+  >>> IHello.names()
+  dict_keys(['hello'])
```

Comment:
That will vary on different versions of Python, won't it? Python 2 still returns a list? Which would result in doctest failures if this file gets run there.

Reply:
Thank you for the feedback. I have to admit, I totally forgot about Python 2 compatibility when revising the document. Coincidentially, I talked with @icemac about bringing documentation under doctest, which is not the case in the moment. And I guess it won't in near future.

As Python 3 is the future, I do not think it is absolutely necessary to bloat the document to be fully Python 2 and 3 compatible.

Unfortunately, it is also no good idea to reference the previous version (from 9 years ago), as the code examples had several bugs.

I think it would be a good idea to add a notice at the top of the document, that is was written for Python 3 only.

Also, one could add something like "If you want to use Python 2, please visit https://zopeinterface.readthedocs.io/en/latest/ " or any specific url.

What's your take on it?